taskcluster/ac_taskgraph/transforms/all_dep.py

```diff
@@ -0,0 +1,48 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Apply some defaults and minor modifications to the single_dep jobs.
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.schema import resolve_keyed_by
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def build_name_and_attributes(config, tasks):
+    for task in tasks:
+        all_dependent_tasks = task.pop("dependent-tasks").values()
+        task["dependencies"] = {
+            dep.label: dep.label
+            for dep in all_dependent_tasks
+        }
+        first_dep = all_dependent_tasks[0]
+        copy_of_attributes = first_dep.attributes.copy()
```

Comment:
It could be named `primary-dependency` but I think this name is misleading: `all_dep` doesn't have any concept of primary, all jobs are the same. It just takes the first one available as an implementation detail.